service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java

```diff
@@ -150,139 +176,136 @@ public void reset(final PolicyEvaluationRequestCacheKey key) {
     public void resetForSubjects(final String zoneId, final List<SubjectEntity> subjectEntities) {
         multisetForSubjects(zoneId,
-                subjectEntities.stream().map(subject -> subject.getSubjectIdentifier()).collect(Collectors.toList()));
+                subjectEntities.stream().map(SubjectEntity::getSubjectIdentifier).collect(Collectors.toList()));
     }
 
     private void multisetForSubjects(final String zoneId, final List<String> subjectIds) {
+        String timestamp = timestampValue();
         Map<String, String> map = new HashMap<>();
         subjectIds.forEach(subjectId -> {
-            String key = subjectKey(zoneId, subjectId);
-            String timestamp = timestampValue();
-            logSetSubjectTimestampDebugMessage(key, timestamp, subjectId);
-            map.put(key, timestamp);
+            String subjectKey = subjectKey(zoneId, subjectId);
+            logSetEntityTimestampsDebugMessage(timestamp, subjectKey, subjectId, EntityType.SUBJECT);
+            map.put(subjectKey, timestamp);
         });
         multiSet(map);
     }
 
-    private void logSetSubjectTimestampDebugMessage(final String key, final String timestamp, final String subjectId) {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug(String.format("Setting timestamp for subject '%s'; key: '%s', value: '%s' ", subjectId, key,
-                    timestamp));
-        }
-    }
-
     private boolean isRequestCached(final List<String> values) {
-        if (null != values.get(values.size() - 1)) {
-            return true;
-        }
-        return false;
+        return null != values.get(values.size() - 1);
     }
 
     private boolean isCachedRequestInvalid(final List<String> values, final DateTime policyEvalTimestamp) {
         DateTime policyEvalTimestampUTC = policyEvalTimestamp.withZone(DateTimeZone.UTC);
 
-        if (connectorService.isResourceAttributeConnectorConfigured()
-                || connectorService.isSubjectAttributeConnectorConfigured()) {
-            return haveConnectorCacheIntervalsLapsed(connectorService, policyEvalTimestampUTC);
+        if (this.connectorService.isResourceAttributeConnectorConfigured()
+                || this.connectorService.isSubjectAttributeConnectorConfigured()) {
+            return haveConnectorCacheIntervalsLapsed(this.connectorService, policyEvalTimestampUTC);
         } else {
             return havePrivilegeServiceAttributesChanged(values, policyEvalTimestampUTC);
         }
     }
 
-    boolean havePrivilegeServiceAttributesChanged(final List<String> values,
-            final DateTime policyEvalTimestampUTC) {
-        for (int i = 0; i < values.size() - 1; i++) {
-            if (null == values.get(i)) {
-                continue;
+    boolean havePrivilegeServiceAttributesChanged(final List<String> values, final DateTime policyEvalTimestampUTC) {
+        for (String value : values) {
+            if (null == value) {
+                return true;
```

Comment:
Values were already tested for nulls by isRequestCached before calling this method.